ibis/backends/pandas/udf.py

```diff
@@ -220,7 +220,20 @@ def pre_execute_analytic_and_reduction_udf(op, *clients, scope=None, **kwargs):
     # 3) an ungrouped custom aggregation context
     @execute_node.register(type(op), *(itertools.repeat(pd.Series, nargs)))
     def execute_udaf_node_no_groupby(op, *args, aggcontext, **kwargs):
-        return aggcontext.agg(args[0], op.func, *args[1:])
+        func = op.func
+
+        def aggregator(first, *rest):
+            result = func(first, *rest)
+
+            # Here we don't user execution.util.coerce_to_output
+            # because this is the inner loop and we do not want
+            # to wrap a scalar value with a series.
+            if isinstance(op._output_type, dt.Struct):
+                return coerce_to_dataframe(result, op._output_type.names)
```

Comment:
This is fine for the purpose of this change. In the long term, I think we do want to consider lift this to outside for performance reasons.

This is not so much an issue for execute_udaf_node_no_groupby since there is a single group. For `execute_udaf_node_groupby`, it does seems wasteful to create a one row pd.DataFrame to represent basically a tuple of scalar. Sth to think about the improve in the future how best to do this.